### DISCUSSION TITLE
What is the relationship with Deployment Environments?

### DISCUSSION THREAD

**Discussion by TWolversonReply** [Azure/azure-dev#2839] [Q&A] [answered]
Today I've stumbled on [Deployment Environments](https://learn.microsoft.com/en-us/azure/deployment-environments/). They seem to be in the same niche as Azd, but don't mention it and don't look like they use any of the same conventions. Are they meant to be complementary? Do they both do the same thing? If so, which should I choose?

**Comment by savannahostrowski** [comment, marked as answer]
We are currently working on support for Azure Deployment Environments in azd (it should be out next month). You can think of azd as the developer interface/tool for interacting with Azure Deployment Environments. Or in other words, you'd use them together. azd would give you the gestures and ADE would give you the project, environment types, environment definitions etc.

**Reply by CMeeg** [reply]
Support for Azure Deployment Environments in azd sounds really interesting - looking forward to this. I was also wondering how the two fit together before I chose to go with creating an azd template (I was also looking at [Bicep deployment stacks](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/deployment-stacks)).

Ultimately I chose azd because I personally wanted a developer tool, but I think being able to leverage the ops tooling (governance, auditing, insights) built into ADE will help adopting azd at my workplace.

**Reply by savannahostrowski** [reply]
That's great to hear! Looking forward to your feedback once it's released!
